core/parentchain/light-client/src/light_validation_state.rs

```diff
@@ -17,39 +17,32 @@
 
 //! State of the light-client validation.
 
-use crate::{state::RelayState, Error, RelayId};
+use crate::state::RelayState;
 use codec::{Decode, Encode};
 pub use sp_finality_grandpa::SetId;
 use sp_runtime::traits::Block as ParentchainBlockTrait;
-use std::collections::BTreeMap;
 
 #[derive(Encode, Decode, Clone, Debug)]
```

Comment:
It was, defined as such:
```
impl<Block: ParentchainBlockTrait> Default for LightValidationState<Block> {
	fn default() -> Self {
		Self { num_relays: Default::default(), tracked_relays: Default::default() }
	}
}
```
And since commit 83d3e4806631aff68f9c132e8337ef0bbe4e6682 
`RelayState<T>` derives `Default` here:
https://github.com/integritee-network/worker/blob/83d3e4806631aff68f9c132e8337ef0bbe4e6682/core/parentchain/light-client/src/state.rs#L26

I think it should be derivable :)

Reply:
This is an interesting catch, especially because I wonder why your line does not generate a compiler error, but only when I add it to the light-validation-state too:

```
   Compiling itc-parentchain-light-client v0.9.0 (/opt/shared/integritee-worker/core/parentchain/light-client)
error[E0277]: the trait bound `<Block as sp_runtime::traits::Block>::Header: Default` is not satisfied
  --> /opt/shared/integritee-worker/core/parentchain/light-client/src/light_validation_state.rs:27:2
   |
25 | #[derive(Encode, Decode, Clone, Debug, Default)]
   |                                        ------- in this derive macro expansion
26 | pub struct LightValidationState<Block: ParentchainBlockTrait> {
27 |     pub(crate) relay_state: RelayState<Block>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `<Block as sp_runtime::traits::Block>::Header`
   |
note: required for `RelayState<Block>` to implement `Default`
  --> /opt/shared/integritee-worker/core/parentchain/light-client/src/state.rs:26:44
   |
26 | #[derive(Encode, Decode, Clone, PartialEq, Default)]
   |                                            ^^^^^^^
   = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Adding a `Default` trait-bound does not make sense because the actual header does indeed not implement `Default` (I double-checked).

Reply:
This is actually my guess too. I will remove it from the `RelayState` as it might give a wrong impression.